tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/AspNet/ExceptionHandlerExtensions_HandleAsync_Integration.cs

```diff
@@ -0,0 +1,79 @@
+        MinimumVersion = Major5Minor1,
+        MaximumVersion = Major5MinorX,
+        IntegrationName = IntegrationName)]
+    [Browsable(false)]
+    [EditorBrowsable(EditorBrowsableState.Never)]
+    public class ExceptionHandlerExtensions_HandleAsync_Integration
+    {
+        private const string Major5Minor1 = "5.1";
+        private const string Major5MinorX = "5";
+        private const string IntegrationName = nameof(IntegrationIds.AspNetWebApi2);
+
+        /// <summary>
+        /// OnMethodBegin callback
+        /// </summary>
+        /// <typeparam name="TTarget">Type of the target</typeparam>
+        /// <typeparam name="TExceptionHandler">Type of the configuration callback action</typeparam>
+        /// <typeparam name="TExceptionContext">Type of the exception context</typeparam>
+        /// <param name="instance">Instance value, aka `this` of the instrumented method. This will be null because the method is static</param>
+        /// <param name="handler">The configured exception handler value</param>
+        /// <param name="context">The exception context value</param>
+        /// <param name="cancellationToken">CancellationToken value</param>
+        /// <returns>Calltarget state value</returns>
+        public static CallTargetState OnMethodBegin<TTarget, TExceptionHandler, TExceptionContext>(TTarget instance, TExceptionHandler handler, TExceptionContext context, CancellationToken cancellationToken)
+            where TExceptionContext : IExceptionContext
+        {
+            var scope = Tracer.Instance.ActiveScope;
+            var exception = context.Exception;
+
+            if (scope is not null && exception is not null)
+            {
+                // Only try setting an exception if there's an active span
+                // The rest of the instrumentation will handle disposing the scope
+                scope.Span.SetException(exception);
+            }
+
+            return CallTargetState.GetDefault();
+        }
+
+        /// <summary>
+        /// OnAsyncMethodEnd callback
+        /// </summary>
+        /// <typeparam name="TTarget">Type of the target</typeparam>
+        /// <typeparam name="TResponse">Type of the response, in an async scenario will be T of Task of T</typeparam>
+        /// <param name="instance">Instance value, aka `this` of the instrumented method. This will be null because the method is static</param>
+        /// <param name="response">Response instance</param>
+        /// <param name="exception">Exception instance in case the original code threw an exception.</param>
+        /// <param name="state">Calltarget state value</param>
+        /// <returns>A response value, in an async scenario will be T of Task of T</returns>
+        public static TResponse OnAsyncMethodEnd<TTarget, TResponse>(TTarget instance, TResponse response, Exception exception, CallTargetState state)
```

Comment:
I believe you need to add `[PreserveContext]` to this function, but @kevingosse will confirm

Reply:
>Kevin: Well the attribute isn't needed if the code in the integration doesn't use the HttpContext. Here it doesn't look like it does

So ignore me 🙂